service/controller/resource/subnet/resource.go

```diff
@@ -152,11 +153,53 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 			return nil
 		}
+
+		if key.IsSucceededProvisioningState(*currentDeployment.Properties.ProvisioningState) {
+			storageAccountsClient, err := r.azureClientsFactory.GetStorageAccountsClient(credentialSecret.Namespace, credentialSecret.Name)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			storageAccount, err := storageAccountsClient.GetProperties(ctx, key.ClusterID(&cr), key.StorageAccountName(&cr), "")
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			if !isSubnetAllowedToStorageAccount(ctx, storageAccount, currentDeployment) {
+				*storageAccount.AccountProperties.NetworkRuleSet.VirtualNetworkRules = append(*storageAccount.AccountProperties.NetworkRuleSet.VirtualNetworkRules, storage.VirtualNetworkRule{VirtualNetworkResourceID: nil})
```

Comment:
I'm probably missing something, but how does this add the current subnet to the allowed subnets lists?